logo.js

```diff
@@ -546,6 +552,32 @@ function LogoInterpreter(turtle, stream, savehook)
     return lhs;
   }
 
+  function defer(func) {
+    var input = Array.prototype.slice.call(arguments, 1);
+    return function () {
```

Comment:
I think this can be simplified to:

```js
return function() {
  return Promise.all(input).then(function(args) { return func.apply(null, args); });
};
```